master/internal/rm/kubernetesrm/pods.go

```diff
@@ -897,10 +924,104 @@ func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
 	ctx.Respond(response)
 }
 
-// summarize will return all nodes currently in the k8 cluster that have GPUs as agents.
-// It will map currently running Determined pods to the slots on these Nodes, marking all other
-// slots as Free, even if they are being used by other k8 pods.
-func (p *pods) summarize(ctx *actor.Context) map[string]model.AgentSummary {
+// summarize describes pods' available resources. When there's exactly one resource pool and that
+// pool has no quotas configured, it uses the whole cluster's info. Otherwise, it uses namespaces'
+// quotas to derive that info.
+func (p *pods) summarize(ctx *actor.Context) (map[string]model.AgentSummary, error) {
+	namespaceToQuota := make(map[string]k8sV1.ResourceQuota)
+
+	// Look up quotas for our resource pools' namespaces
+	for namespace := range p.namespaceToPoolName {
+		quotaList, err := p.quotaInterfaces[namespace].List(context.TODO(), metaV1.ListOptions{})
+		if k8serrors.IsNotFound(err) || quotaList == nil || len(quotaList.Items) != 1 {
+			// TODO: figure out how we want to handle multiple quotas per namespace?
+			continue
+		} else if err != nil {
+			return nil, err
+		}
+
+		namespaceToQuota[namespace] = quotaList.Items[0]
+	}
+
+	// If there's only one resource pool configured and it doesn't have a quota, summarize using the
+	// whole cluster.
+	if len(p.namespaceToPoolName) == 1 {
+		var namespaceOfPool string
+		for namespace := range p.namespaceToPoolName {
+			namespaceOfPool = namespace
+		}
+
+		// If there's no quota for our only resource pool's namespace
+		if _, ok := namespaceToQuota[namespaceOfPool]; !ok {
+			return p.summarizeCluster(ctx), nil
+		}
+	}
+
+	containers := p.containersPerResourcePool()
+	summaries := make(map[string]model.AgentSummary, len(p.namespaceToPoolName))
+	for namespace, poolName := range p.namespaceToPoolName {
+		quota, quotaExists := namespaceToQuota[namespace]
+		numContainers := containers[poolName]
+
+		var (
+			registeredTime time.Time
+			slots          map[string]model.SlotSummary
+		)
+		if quotaExists {
```

Comment:
I don't feel too strongly, but this looks more natural to me:
```suggestion
		slots := model.SlotsSummary{}
		numContainers := containers[poolName]
		var registeredTime time.Time
		if quota, quotaExists := namespaceToQuota[namespace]; quotaExists {
```